packages/flutter/lib/src/material/navigation_bar.dart

```diff
@@ -230,6 +257,9 @@ class NavigationDestination extends StatelessWidget {
     this.selectedIcon,
     required this.label,
     this.tooltip,
+    this.showBadge = false,
```

Comment:
This doesn't seem like a natural Flutter API. Why not give NavigationDestination a `Widget?` valued badge property and provide a standard NavigationBadge widget with the right default colors, shape etc?

Reply:
Thanks for the feedback. 

I've reworked Badge to be an independent widget and will be supported via `Widget? badge` parameter in `NavigationDestination` and `NavigationRailDestination` in separate PRs.